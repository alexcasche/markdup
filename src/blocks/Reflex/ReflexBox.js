import styled from 'styled-components'
import { applyStyleModifiers } from 'styled-components-modifiers'
import { ReflexElement } from 'react-reflex'

const MODIFIERS = {
  expanded: () => `
    position: fixed !important;
    top: 0;
    left: 0;
    z-index: 999;
  `,
}

const ReflexBox = styled(ReflexElement)`
  overflow: hidden;
  @media only screen and (max-width: ${props =>
      props.theme.breakPoints.mobile}) {
    width: 100vw !important;
  }
  ${applyStyleModifiers(MODIFIERS)};
`

export default ReflexBox
