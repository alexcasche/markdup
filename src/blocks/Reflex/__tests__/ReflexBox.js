import ReflexBox from '../ReflexBox'

describe('ReflexBox Element', () => {
  it('renders correctly', () => {
    const tree = render(<ReflexBox />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('adds expanded modifier', () => {
    const tree = render(<ReflexBox modifiers={['expanded']} />).toJSON()
    expect(tree).toMatchSnapshot()
    expect(tree).toHaveStyleRule('z-index', '999')
  })
})
