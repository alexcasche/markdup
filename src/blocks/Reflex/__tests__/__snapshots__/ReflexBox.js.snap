// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReflexBox Element adds expanded modifier 1`] = `
.c0 {
  overflow: hidden;
  position: fixed !important;
  top: 0;
  left: 0;
  z-index: 999;
}

@media only screen and (max-width:600px) {
  .c0 {
    width: 100vw !important;
  }
}

<div
  className="reflex-element c0"
  style={
    Object {
      "FlexElement": undefined,
      "MozBoxFlex": undefined,
      "WebkitBoxFlex": undefined,
      "WebkitFlex": undefined,
      "flex": undefined,
    }
  }
>
  <div
    style={
      Object {
        "height": "100%",
        "width": "100%",
      }
    }
  />
</div>
`;

exports[`ReflexBox Element renders correctly 1`] = `
.c0 {
  overflow: hidden;
}

@media only screen and (max-width:600px) {
  .c0 {
    width: 100vw !important;
  }
}

<div
  className="reflex-element c0"
  style={
    Object {
      "FlexElement": undefined,
      "MozBoxFlex": undefined,
      "WebkitBoxFlex": undefined,
      "WebkitFlex": undefined,
      "flex": undefined,
    }
  }
>
  <div
    style={
      Object {
        "height": "100%",
        "width": "100%",
      }
    }
  />
</div>
`;
